### PR TITLE
Dart flexible sync

### PR DESCRIPTION
**Pull request by blagoev**

### CODE REVIEW

File: dart_flexible_sync/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+# Realm Dart SDK
+
+The Realm Dart package is `realm_dart` and it is available in [pub.dev](https://pub.dev/packages/realm_dart)
+
+## Environment setup for Realm Dart
+
+* Supported platforms are Windows, Mac and Linux.
+
+* Dart SDK ^2.17.1
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:20]: We do actually support 2.17.0 currently

File: dart_flexible_sync/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+## A simple command-line application using [Realm Dart SDK](https://www.mongodb.com/docs/realm/sdk/flutter/#dart-standalone-realm) Flexible Sync with [Atlas App Services application](https://www.mongodb.com/docs/atlas/app-services/).
+This sample demonstrates the usage of Flexible Sync with Realm Dart SDK. 
+Writing to a local realm sends the data automatically to the Atlas Cloud Collection.
+Then the local realm is subscribed only for items matching specific query criterias.
+This causes the local realm to be populated only with specific items matching the filter. 
+All other items will exist only in Atlas cloud collection.
+
+# Realm Dart SDK
+
+The Realm Dart package is `realm_dart` and it is available in [pub.dev](https://pub.dev/packages/realm_dart)
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:14]:
```suggestion
The Realm Dart package is `realm_dart` and it is available on [pub.dev](https://pub.dev/packages/realm_dart)
```
I think

File: dart_flexible_sync/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## Environment setup for Realm Dart
+
+* Supported platforms are Windows, Mac and Linux.
+
+* Dart SDK ^2.17.1
+
+# Atlas App Services Configuration Steps
+
+* Create an account in [realm.mongodb.com](https://realm.mongodb.com) - follow the instructions: [Get Started with Atlas](https://www.mongodb.com/docs/atlas/getting-started)
+* Creat a new app following the instruction: [Create an App with Atlas App Services UI](https://www.mongodb.com/docs/atlas/app-services/manage-apps/create/create-with-realm-ui).
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:25]:
```suggestion
* Create a new app following the instructions here: [Create an App with Atlas App Services UI](https://www.mongodb.com/docs/atlas/app-services/manage-apps/create/create-with-realm-ui).
```

File: dart_flexible_sync/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    For the purpose of this sample you don't need to create an app from a template. You can just create an empty application.
+* Click the button in the blue line above - `Review draft & deploy`.
+* Go to `Authentication Providers` menu from the left panel and make sure the option "Allow users to log in anonymously" is ON.
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:28]:
```suggestion
* Go to the `Authentication Providers` menu in the left panel and make sure the option "Allow users to log in anonymously" is ON.
```
I think

File: dart_flexible_sync/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    Read [this page](https://www.mongodb.com/docs/atlas/app-services/authentication/providers/) for more information about the other types of authentication.
+* Go to `Sync` menu and [Enable Flexible Sync](https://www.mongodb.com/docs/atlas/app-services/sync/configure/enable-sync/#enable-flexible-sync).
+    * You can skip button "Create a Schema". Choose "No thanks, continue to Sync".
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:31]:
```suggestion
    * Don't create a schema. Skip by choosing "No thanks, continue to Sync".
```
I think we should ensure the user don't think they can choose their own schema here

File: dart_flexible_sync/README.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    * Please choose button "Flexible Sync". Only Flexible Sync is supported in Realm Dart SDK.
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:32]:
```suggestion
    * Press the "Flexible Sync" button. Only Flexible Sync is supported in the Realm Dart SDK.
```
It is an order :-)

File: dart_flexible_sync/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    * Switch ON ["Development mode"](https://www.mongodb.com/docs/atlas/app-services/sync/data-model/development-mode/) option.
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:33]:
```suggestion
    * Switch ON the ["Development mode"](https://www.mongodb.com/docs/atlas/app-services/sync/data-model/development-mode/) option. 
```
I think

File: dart_flexible_sync/README.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    For the purpose of this sample you don't need to create an app from a template. You can just create an empty application.
+* Click the button in the blue line above - `Review draft & deploy`.
+* Go to `Authentication Providers` menu from the left panel and make sure the option "Allow users to log in anonymously" is ON.
+    Read [this page](https://www.mongodb.com/docs/atlas/app-services/authentication/providers/) for more information about the other types of authentication.
+* Go to `Sync` menu and [Enable Flexible Sync](https://www.mongodb.com/docs/atlas/app-services/sync/configure/enable-sync/#enable-flexible-sync).
+    * You can skip button "Create a Schema". Choose "No thanks, continue to Sync".
+    * Please choose button "Flexible Sync". Only Flexible Sync is supported in Realm Dart SDK.
+    * Switch ON ["Development mode"](https://www.mongodb.com/docs/atlas/app-services/sync/data-model/development-mode/) option. 
+    * Choose to create a new database collection and set a name of this database.
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:34]:
```suggestion
    * Create a new database collection and choose a name for this.
```

File: dart_flexible_sync/README.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    * Create a new queryable field used for filtering data between both realms in this application. For our sample the fields are `status` and `progressMinutes`. 
+        Since it is not available in the selection list, just start writing inside the selection box and then choose `Create status`. 
+        It will be created.
+        Do the same for field `progressMinutes`.
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:38]:
```suggestion
        Do the same for the field `progressMinutes`.
```

File: dart_flexible_sync/README.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    * Define permission - for the purpose of this sample please choose the optionn `Users can read and write all data`.
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:39]:
```suggestion
    * Define permission - for the purpose of this sample choose the option `Users can read and write all data`.
```

File: dart_flexible_sync/README.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    * Click the button `Enable Sync` and confirm.
+    * Click again the button in the blue line above - `Review draft & deploy`.
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:41]:
```suggestion
    * Click the button in the blue line above - `Review draft & deploy`, again.
```

File: dart_flexible_sync/README.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+* Copy App ID of your new application. Follow [Find an App ID](https://www.mongodb.com/docs/atlas/app-services/reference/find-your-project-or-app-id/?_ga=2.267445390.1225817147.1654079983-1571915642.1647002315&_gac=1.229371374.1654173423.CjwKCAjwv-GUBhAzEiwASUMm4jBtzETN-YJq0KELgeGLKk-4_6wVAfImtPoBbo-A35_eKjZ1p0Lh_BoCotcQAvD_BwE#find-an-app-id)
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:42]:
```suggestion
* Copy the App ID of your new application. Follow [Find an App ID](https://www.mongodb.com/docs/atlas/app-services/reference/find-your-project-or-app-id/?_ga=2.267445390.1225817147.1654079983-1571915642.1647002315&_gac=1.229371374.1654173423.CjwKCAjwv-GUBhAzEiwASUMm4jBtzETN-YJq0KELgeGLKk-4_6wVAfImtPoBbo-A35_eKjZ1p0Lh_BoCotcQAvD_BwE#find-an-app-id)
```

File: dart_flexible_sync/README.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    ```
+    dart pub add realm_dart
+    ```
+
+* Install the `realm_dart` package into the application. This downloads and copies the required native binaries to the app directory.
+
+    ```
+    dart run realm_dart install
+    ``` 
+
```
Comment (nielsenko) [inline dart_flexible_sync/README.md:64]: Should this be here, and if it should, then what about `dart run realm_dart generate`?  Perhaps better to just refer to the realm_dart README

File: dart_flexible_sync/pubspec.yaml [hunk 14]
```diff
@@ -0,0 +1,14 @@
+name: tasktracker
+description: MongoDB flexible sync example.
+version: 1.0.0
+
+environment:
+  sdk: ">=2.17.1 <3.0.0"
```
Comment (nielsenko) [inline dart_flexible_sync/pubspec.yaml:6]: Could loosen this to >=2.17.0..

**Review by nielsenko — COMMENTED** [review]
Looks good 👍 

I have some suggestions. English is not my native tongue, so if they feel odd to you, please consult a better expert 😊

File: dart_flexible_sync/README.md (hunk 13 above)
Comment (desistefanova) [reply dart_flexible_sync/README.md:64]: Good catch. We need only `dart run realm_dart install`

**Review by nielsenko — APPROVED** [review]

**Pull request merged**